airflow/models/mappedoperator.py

```diff
@@ -559,7 +562,12 @@ def _get_map_lengths(self, run_id: str, *, session: Session) -> Dict[str, int]:
         for task_id, length in xcom_query:
             for mapped_arg_name in mapped_dep_keys[task_id]:
                 map_lengths[mapped_arg_name] += length
+        return map_lengths
 
+    def _resolve_map_lengths(self, run_id: str, *, session: Session) -> Dict[str, int]:
+        """Return dict of argument name to map length, or throw if some are not resolvable"""
+        expansion_kwargs = self._get_expansion_kwargs()
+        map_lengths = self._get_map_lengths(run_id, session=session)
```

Comment:
We should document `_get_map_lengths` to explain how it’s different from `_resolve_map_lengths`.

Reply:
Done:

>         """Return dict of argument name to map length.
>
>        If any arguments are not known right now (upstream task not finished) they will not be present in the
>        dict.